middleware/preview.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+const { getSecret } = require('../modules/secrets');
+
+// Sets a global parameter to indicate if this request came from a preview domain
+module.exports = (req, res, next) => {
+    let PREVIEW_DOMAIN = process.env.PREVIEW_DOMAIN || getSecret('preview.domain');
```

Comment:
This could happen outside of the middleware so we're not looking this up on every request.

Reply:
Done!